app.py

```diff
@@ -0,0 +1,75 @@
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel
+from typing import List
+from transformers import pipeline
+import os
+
+app = FastAPI()
+# Initialize the NSFW model once
+env = os.environ.get("env", "dev")
+torch_device = "cuda" if env == "prod" else "cpu" 
+
+class InferenceData(BaseModel):
+    name: str
+    shape: List[int]
+    data: List
+    datatype: str
+
+class InputRequest(BaseModel):
+    inputs: List[InferenceData]
+
+class OutputResponse(BaseModel):
+    modelname: str
+    modelversion: str
+    outputs: List[InferenceData]
+
+@app.get("/")
+async def hello_world():
+    return "nsfw_text"
+
+@app.post("/validate", response_model=OutputResponse)
+async def check_nsfw(input_request: InputRequest):
+    threshold = None
+    for inp in input_request.inputs:
+        if inp.name == "text":
+            text_vals = inp.data
+        elif inp.name == "threshold":
+            threshold = float(inp.data[0])
+
+    if text_vals is None or threshold is None:
+        raise HTTPException(status_code=400, detail="Invalid input format")
+
+    return NSFWText.infer(text_vals, threshold)
+
+class NSFWText:
+    model_name = "michellejieli/NSFW_text_classifier"
```

Comment:
We're NOT using detoxify here?

Reply:
IIRC the new pattern was to specifically use the underlying model and not use the validator in the hosted fast api so that we're getting the raw inference response in the validator client side.